### ISSUE TITLE
[Week 4] Implement Attention Model and Multiscale Fusion

### ISSUE THREAD

**Issue by IamArmanNikkhah** [IamArmanNikkhah/sae-viewport-prediction#10]
### Context
This task completes the feature engineering "Track A". We need to implement two critical components of the SAE metric:
1.  **Attention Model:** Extrapolating where the user *should* be looking based on current velocity.
2.  **Multiscale Fusion:** Aggregating estimators with different timescales ($\tau$) into a single score.

### Tasks
- [ ] **Attention Model (`src/sae_core/attention.py`):**
    - [ ] Implement Rodrigues' rotation formula to compute $\nu_t = R(\omega_t \Delta) x_t$.
    - [ ] Implement motion-adaptive concentration: $\lambda_s = \lambda_{base} (1 + \gamma \|\omega_t\|)$.
    - [ ] *Note:* $\gamma$ is a hyperparameter (try 0.1). $\lambda_{base}$ can be fixed (try 10.0) or looked up from a table.
- [ ] **Multiscale Fusion (`src/sae_core/fusion.py`):**
    - [ ] Implement Fisher Proxy calculation: $I_s = N_s \kappa_s A_3(\kappa_s)$.
    - [ ] Implement Stabilized Softmax to compute weights $\alpha_s$ from $I_s$.
    - [ ] Compute final aggregation: $SAE(t) = \sum \alpha_s SAE_s(t)$.

### Acceptance Criteria
- [ ] `attention_direction` function correctly rotates a vector given an axis and angle.
- [ ] Fusion weights $\alpha_s$ sum to 1.0.
- [ ] The system outputs a single scalar $SAE(t)$ for a given time step.

**Labels:** `week-4`, `feature-engineering`
**Assignee:** (Assign to "Track A" team members)

**Comment by VenkataMudunuri** [comment]
After conducting a small test to verify whether the attention_direction () function in the attention.py file is actually able to rotate a vector or not. Whether or not, the motion_adaptive_concentration() function produces realistic λₛ values, and the fuse_multiscale_sae () function is able to compute we the weights that at the end sum to 1 and can return 1 single scalar SAE(t). 

This is the self-check results for if attention.py is going smoothly:

<img width="446" height="46" alt="Image" src="https://github.com/user-attachments/assets/543f30d9-8d37-4b4e-a810-b305819e884d" />

This is the self-check for if fusion.py is going smoothly:

<img width="482" height="120" alt="Image" src="https://github.com/user-attachments/assets/ea077eee-0798-4530-8c26-c08184ed8e45" />
